lib/Doctrine/Connection/Statement.php

```diff
@@ -271,9 +271,11 @@ public function execute($params = null)
             $this->_conn->getListener()->postStmtExecute($event);
 
             //fix a possible "ORA-01000: maximum open cursors exceeded" when many non-SELECTs are executed and the profiling is enabled
-            if (strtoupper(substr($this->_stmt->queryString,0,6)) != 'SELECT' && strtoupper(substr($this->_stmt->queryString,0,4)) != 'WITH' ){
+            $queryBeginningSubstring = strtoupper(substr(ltrim($this->_stmt->queryString), 0, 6));
+            if (substr($queryBeginningSubstring, 0, 6) != 'SELECT' && substr($queryBeginningSubstring, 0, 4) != 'WITH' ){
```

Comment:
The first `substr` isn't necessary since it was performed just before.
But we need to keep the second one.